File: leetcode/python/0070_climbing_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [1, 2]
        if n == 1: return dp[0]
        if n == 2: return dp[1]

        for i in range(3, n + 1):
            tmp = dp[1]
            dp[1] = dp[0] + dp[1]
            dp[0] = tmp

        return dp[1] 

File: leetcode/java/0070_climbing_stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n + 1];
        dp[1] = 1;

        if (n >= 2) dp[2] = 2;

        for (int i = 3; i < dp.length; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }

        return dp[dp.length - 1];
    }
}